src/types.ts

```diff
@@ -318,13 +318,19 @@ export function valueTypeName(field: FieldDescriptorProto): TypeName {
 export function messageToTypeName(
   typeMap: TypeMap,
   protoType: string,
+  options: (Options | null) = null,
```

Comment:
Hrm, can we make `options` required? I'd be worried about a caller getting back the wrong type name b/c they weren't passing in options... If this is a huge hassle, an in-source todo/separate follow up would be fine.

Reply:
We can. The only callsites of `messageToTypeName` that don't currently have access to `options` are `requestType` and `responseType`. We could pass `options` into those, though, I think. I was just shying away from a bigger change.